drivers/src/nrfx_nfct.c

```diff
@@ -625,15 +625,15 @@ nrfx_err_t nrfx_nfct_nfcid1_default_bytes_get(uint8_t * const p_nfcid1_buff,
         return NRFX_ERROR_INVALID_LENGTH;
     }
 
-#if defined(FICR_NFC_TAGHEADER0_MFGID_Msk) && !defined(NRF_TRUSTZONE_NONSECURE)
+#if defined(NRF_TRUSTZONE_NONSECURE)
+    /* The non-secure domain does not have access to NRF_FICR. It must
+    read them indirectly through the secure read service instead. */
+    return NRFX_ERROR_INVALID_STATE;
+#endif
+
     uint32_t nfc_tag_header0 = NRF_FICR->NFC.TAGHEADER0;
     uint32_t nfc_tag_header1 = NRF_FICR->NFC.TAGHEADER1;
     uint32_t nfc_tag_header2 = NRF_FICR->NFC.TAGHEADER2;
```

Comment:
These three lines cannot be compiled for a non-secure image, since `NRF_FICR` is not defined then. Therefore, you should rather use `#else` above and move that `#endif` to the end of this function.

Reply:
Done